### PR TITLE
Add script for checking files changed in a PR

### PR DESCRIPTION
**Pull request by ndodda-amazon**
### Description of changes:
This script is used to conditionally run profiler integration tests in the CI in two ways:
- Builds will automatically check files changed in a PR and only run tests if relevant files were changed
- Developers can manually set the `disable_integration_tests` environment variable in this script so that profiler integration tests are not run in the CI. This meant to be a temporary tool for speeding up development - **the environment variable must be reset to `"true"` before approving and merging a PR**. 

See [this PR](https://tiny.amazon.com/21wpjr6s/IsenLink) for more details, which is where this script will be used.

Also cut down on code duplication by:
- Refactoring buildspecs by only populating them with environment variables and calling a single shell script for running the profiler integration tests. 
- Moving dependent packages to `requirements.txt` so there's a single file that manages packages.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by NihalHarish** [comment]
Can you add enable the integration test codebuild job as part of this PR?

**Comment by ndodda-amazon** [comment]
> Can you add enable the integration test codebuild job as part of this PR?

I can't enable it for just one PR, I'd have to enable it for all PRs in the repo. Are we okay with doing that? I've already done 10 manual tests (as seen in the other PR's description) so I'm confident there won't be issues.

**Comment by ndodda-amazon** [comment]
> Can you add enable the integration test codebuild job as part of this PR?

Added. As you can see, the TF 2.3 and PT 1.6 builds pass trivially because they skip the integration tests (only files in the `config` folder are modified in this PR)

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml:6]: maybe this variable can be renamed?

The comment seems to suggest that toggling this variable changes the version of smdebug involved.

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml:32]: +1

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:6]: maybe this variable can be renamed?

The comment seems to suggest that toggling this variable changes the version of smdebug involved.

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:17]: I wonder if we can simply manage trivial dependencies like pip with a single file or env variable.

We can hold this thought for later; but it is clear that managing versions across multiple files is a certain pain point.

**Review by NihalHarish — COMMENTED** [review]
Looks good to me!
We can approve it after you address the single comment I have made about the the variable name.

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml:6]: Sure, can rename to `override_smdebug_version`

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:17]: There's this tool called dependabot I've used before that will automatically make pull requests for out of date dependencies. https://dependabot.com/

I can set this up in all of our repos. That way we won't have to spend time doing this ourselves.

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:6]: Sure, can rename to `override_smdebug_version`

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml:34]: Combine this into a single `if block`

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:34]: Combine this into a single if block

File: config/buildspec_profiler_sagemaker_pytorch_1_7_0_integration_tests.yml
Comment (NihalHarish) [inline config/buildspec_profiler_sagemaker_pytorch_1_7_0_integration_tests.yml:33]: combine this into a single if block

**Review by NihalHarish — CHANGES_REQUESTED** [review]

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (NihalHarish) [reply config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml:17]: Awesome!

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml]: Done.

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml]: Done.

File: config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_5_0_integration_tests.yml]: Done.

File: config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_6_0_integration_tests.yml]: Done.

File: config/buildspec_profiler_sagemaker_pytorch_1_7_0_integration_tests.yml
Comment (ndodda-amazon) [reply config/buildspec_profiler_sagemaker_pytorch_1_7_0_integration_tests.yml]: Done.

**Comment by NihalHarish** [comment]
Let's see how we can reduce code duplication across YAML files.

See https://stackoverflow.com/questions/14184971/more-complex-inheritance-in-yaml

**Comment by ndodda-amazon** [comment]
> See https://stackoverflow.com/questions/14184971/more-complex-inheritance-in-yaml

As discussed offline, this solution won't work since we want to have separate buildspecs for each build. However, we can cut down on the duplicated code by moving all of the commands to a shell script and only running that shell script within the buildspec.

Also moved the dependent packages to a requirements.txt, which will be easy to update moving forward. We should probably take the same approach in all of our other repos.

**Review by NihalHarish — APPROVED** [review]

File: config/profiler/run_profiler_integration_tests.sh
Comment (ndodda-amazon) [reply config/profiler/run_profiler_integration_tests.sh:74]: TF DLC doesn't have the `tensorflow-datasets`, but we use that package for our tests. Will add a comment.

File: config/profiler/requirements.txt
Comment (ndodda-amazon) [reply config/profiler/requirements.txt]: Some of them are (sagemaker, flaky, pytest-xdist). We could still move location of files so that eventually all buildspecs can use one requirements.txt? But I think it's best to have separate requirements.txt files if the actual dependencies are different.

File: config/profiler/run_profiler_integration_tests.sh
Comment (NihalHarish) [reply config/profiler/run_profiler_integration_tests.sh:74]: Interesting. Why do they not have it?

Comment (ndodda-amazon) [reply config/profiler/run_profiler_integration_tests.sh:74]: Unclear, but we can start that discussion again.

**Pull request merged**
